vivisect/symboliks/common.py

```diff
@@ -237,7 +237,7 @@ def _solve(self, emu=None, vals=None):
         '''
         raise Exception('%s *must* implement solve(emu=emu)!' % self.__class__.__name__)
 
-    def reduce(self, emu=None, foo=False):
+    def reduce(self, emu=None, foo=True):
```

Comment:
what is this `foo` argument?

i guess it is used, but the name has no meaning to me (almost by purpose?).

Reply:
wow, for a minute i thought i was to blame.  that variable dates back to 4/1/2014, and it belongs to someone else (according to hg blame from that time-period.  yay for once it wasn't my fault!).  it has survived since its inception, where reduction started to run repeatedly until reduction didn't change the outcome of the ast.

it probably should have been something like `repeat_until_done` but that's a long name.  it's been around 7 years, does it make sense to change the api now?  how many tools would break?

Reply:
So it's been around for a while. To keep some level of backwards compatibility, I didn't change the name. But what it does is run the walkTree/reduction stuff until  the strings don't change, meaning that full reduction has finished. 99% of the time, only one reduction pass is really needed, but I have noted that there are some weird corner cases that don't get fully reduced on the first pass.

However,, I hate the name and the fact that it's false by default. If we ever do move to a 2.0.0, I'll probably end up changing the name, because `foo` is an awful parameter name.

Reply:
agreed